File: eval/corpus/disambig/csv_codec.py
import csv
import io


def encode(rows):
    """Serialize a list of row mappings to CSV text with a header line."""
    buf = io.StringIO()
    writer = csv.DictWriter(buf, fieldnames=list(rows[0].keys()))
    writer.writeheader()
    writer.writerows(rows)
    return buf.getvalue()


def csv_marker():
    """Identity marker so callers can establish a csv_codec import edge."""
    return "csv"

File: eval/corpus/disambig/csv_codec.ts
// CSV table codec.
export function encode(rows: Record<string, unknown>[]): string {
  // Serialize a list of table rows to CSV text with a header line and comma-separated columns.
  const header = Object.keys(rows[0]).join(",");
  const body = rows.map((row) => Object.values(row).join(",")).join("\n");
  return header + "\n" + body;
}

export function csvMarker(): string {
  return "csv";
}
